Assign a JPMS module name

The spec API does not have a defined JPMS module name.

I would recommend defining an automatic module name of `org.eclipse.microprofile.context`, since that is the name of the top-level package of this project.

I can't seem to build the project locally (maybe due to incompatible Maven versions?) so go ahead.

```
[ERROR] Failed to execute goal net.revelc.code:impsort-maven-plugin:1.4.1:sort (sort-imports) on project microprofile-context-propagation-api: Execution sort-imports of goal net.revelc.code:impsort-maven-plugin:1.4.1:sort failed: A required class was missing while executing net.revelc.code:impsort-maven-plugin:1.4.1:sort: null
[ERROR] -----------------------------------------------------
[ERROR] realm =    plugin>net.revelc.code:impsort-maven-plugin:1.4.1
[ERROR] strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
[ERROR] urls[0] = file:/Users/david/.m2/repository/net/revelc/code/impsort-maven-plugin/1.4.1/impsort-maven-plugin-1.4.1.jar
[ERROR] urls[1] = file:/Users/david/.m2/repository/com/github/javaparser/javaparser-core/3.15.21/javaparser-core-3.15.21.jar
[ERROR] Number of foreign imports: 1
[ERROR] import: Entry[import  from realm ClassRealm[maven.api, parent: null]]
[ERROR] 
[ERROR] -----------------------------------------------------
[ERROR] : NoClassDefFoundError: org/codehaus/plexus/util/DirectoryScanner: org.codehaus.plexus.util.DirectoryScanner
[ERROR] -> [Help 1]
```

I'm getting an error after adding that `module-info.java` file:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.7.0:jar (attach-javadocs) on project microprofile-context-propagation-api: MavenReportException: Error while generating Javadoc: 
[ERROR] Exit code: 2
[ERROR] error: No source files for package org.eclipse.microprofile.context
[ERROR] 1 error
[ERROR] Command line was: /home/stephane/.sdkman/candidates/java/25-open/bin/javadoc @options @packages
[ERROR] 
[ERROR] Refer to the generated Javadoc files in '/home/stephane/src/java-eclipse/microprofile-context-propagation/api/target/apidocs' dir.
```

--- a/api/pom.xml
+++ b/api/pom.xml
@@ -28,4 +28,19 @@
     <name>MicroProfile Context Propagation</name>
     <description>MicroProfile Context Propagation :: API</description>
 
+    <build>
+        <pluginManagement>
+            <plugins>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-checkstyle-plugin</artifactId>
+                    <configuration>
+                        <!-- https://github.com/checkstyle/checkstyle/issues/8240 -->
+                        <excludes>**/module-info.java</excludes>
+                    </configuration>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+    </build>
+
 </project>

--- a/api/src/main/java/module-info.java
+++ b/api/src/main/java/module-info.java
@@ -1,0 +1,24 @@
+/*
+ * Copyright (c) 2018,2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+module org.eclipse.microprofile.context {
+    exports org.eclipse.microprofile.context;
+    exports org.eclipse.microprofile.context.spi;
+
+    uses org.eclipse.microprofile.context.spi.ContextManagerProvider;
+}

--- a/api/src/main/java/org/eclipse/microprofile/context/ManagedExecutor.java
+++ b/api/src/main/java/org/eclipse/microprofile/context/ManagedExecutor.java
@@ -36,7 +36,7 @@ import org.eclipse.microprofile.context.spi.ContextManagerProvider;
  * <p>
  * Example usage:
  * </p>
- * 
+ *
  * <pre>
  * <code>ManagedExecutor executor = ManagedExecutor.builder().propagated(ThreadContext.APPLICATION).cleared(ThreadContext.ALL_REMAINING).build();
  * ...
@@ -58,7 +58,7 @@ import org.eclipse.microprofile.context.spi.ContextManagerProvider;
  * <p>
  * Example of single action with context propagation:
  * </p>
- * 
+ *
  * <pre>
  * CompletableFuture&lt;?&gt; future = executor
  *    .runAsync(runnable1)
@@ -122,7 +122,7 @@ import org.eclipse.microprofile.context.spi.ContextManagerProvider;
  * defined by EE Concurrency, must raise IllegalStateException upon invocation of the aforementioned life cycle methods,
  * in order to preserve compatibility with that specification.
  * </p>
- * 
+ *
  * <p>
  * Managed executors can forward all contextualised async tasks to a backing executor service if set with
  * {@link ContextManager.Builder#withDefaultExecutorService(ExecutorService)}. Otherwise, a new backing executor service
@@ -147,7 +147,7 @@ public interface ManagedExecutor extends ExecutorService {
      * <p>
      * Example usage:
      * </p>
-     * 
+     *
      * <pre>
      * <code> ManagedExecutor executor = ManagedExecutor.builder()
      *                                                  .maxAsync(5)
@@ -502,7 +502,7 @@ public interface ManagedExecutor extends ExecutorService {
     /**
      * Returns a <code>ThreadContext</code> which has the same propagation settings as this
      * <code>ManagedExecutor</code>, which uses this <code>ManagedExecutor</code> as its default executor.
-     * 
+     *
      * @return a ThreadContext with the same propagation settings as this ManagedExecutor.
      */
     ThreadContext getThreadContext();

--- a/api/src/main/java/org/eclipse/microprofile/context/ThreadContext.java
+++ b/api/src/main/java/org/eclipse/microprofile/context/ThreadContext.java
@@ -40,7 +40,7 @@ import org.eclipse.microprofile.context.spi.ContextManagerProvider;
  * <p>
  * Example usage:
  * </p>
- * 
+ *
  * <pre>
  * <code>ThreadContext threadContext = ThreadContext.builder()
  *     .propagated(ThreadContext.SECURITY, ThreadContext.APPLICATION)
@@ -77,7 +77,7 @@ public interface ThreadContext {
      * <p>
      * Example usage:
      * </p>
-     * 
+     *
      * <pre>
      * <code> ThreadContext threadContext = ThreadContext.builder()
      *                                                   .propagated(ThreadContext.APPLICATION, ThreadContext.SECURITY)
@@ -226,7 +226,7 @@ public interface ThreadContext {
          * For example, to run under the transaction of the thread of execution, with security context cleared and all
          * other thread contexts propagated:
          * </p>
-         * 
+         *
          * <pre>
          * <code> ThreadContext threadContext = ThreadContext.builder()
          *                                                   .unchanged(ThreadContext.TRANSACTION)
@@ -284,9 +284,9 @@ public interface ThreadContext {
      * Identifier for CDI context. CDI context controls the availability of CDI scopes. An empty/default CDI context
      * means that the thread does not have access to the scope of the session, request, and so forth that created the
      * contextualized action.
-     * 
+     *
      * For example, consider the following <code>&#64;RequestScoped</code> resource:
-     * 
+     *
      * <pre>
      * <code>
     &#64;RequestScoped
@@ -297,20 +297,20 @@ public interface ThreadContext {
     }
     </code>
      * </pre>
-     * 
+     *
      * CDI context propagation includes request, session and conversation contexts. When CDI context is propagated, all
      * of the above mentioned contexts that are currently active will be available to the contextualized task with
      * preserved state.
-     * 
+     *
      * <pre>
      * <code>
     ManagedExecutor exec = ManagedExecutor.builder()
         .propagated(ThreadContext.CDI, ThreadContext.APPLICATION)
         .build();
-    
+
     &#64;Inject
     MyRequestScopedBean requestBean;
-    
+
     &#64;GET
     public void foo() {
         exec.supplyAsync(() -&gt; {
@@ -325,7 +325,7 @@ public interface ThreadContext {
      *
      * If CDI context is 'cleared', currently active contexts will still be available to the contextualized task, but
      * their state will be erased.
-     * 
+     *
      * If CDI context is 'unchanged', access to CDI bean's contextual state will be non-deterministic. Namely, context
      * may be missing, or context from a different task may be applied instead. This option is discouraged, and only
      * should be used if CDI context is not used in an application.
@@ -350,13 +350,13 @@ public interface ThreadContext {
      * <p>
      * When using MicroProfile Config to specify defaults, the value <code>None</code> indicates an empty array. For
      * example,
-     * 
+     *
      * <pre>
      * mp.context.ThreadContext.unchanged = None
      * </pre>
-     * 
+     *
      * or
-     * 
+     *
      * <pre>
      * mp.context.ManagedExecutor.propagated = None
      * </pre>
@@ -366,7 +366,7 @@ public interface ThreadContext {
     /**
      * Identifier for security context. Security context controls the credentials that are associated with the thread.
      * An empty/default security context means that the thread is unauthenticated.
-     * 
+     *
      * @see ManagedExecutor.Builder#cleared
      * @see ManagedExecutor.Builder#propagated
      * @see ThreadContext.Builder
@@ -396,7 +396,7 @@ public interface ThreadContext {
      * <p>
      * Example usage:
      * </p>
-     * 
+     *
      * <pre>
      * <code>Executor contextSnapshot = threadContext.currentContextExecutor();
      * ...
@@ -420,7 +420,7 @@ public interface ThreadContext {
      * Wraps a <code>Callable</code> with context that is captured from the thread that invokes
      * <code>contextualCallable</code>.
      * </p>
-     * 
+     *
      * <p>
      * When <code>call</code> is invoked on the proxy instance, context is first established on the thread that will run
      * the <code>call</code> method, then the <code>call</code> method of the provided <code>Callable</code> is invoked.
@@ -467,7 +467,7 @@ public interface ThreadContext {
      * Wraps a <code>Consumer</code> with context that is captured from the thread that invokes
      * <code>contextualConsumer</code>.
      * </p>
-     * 
+     *
      * <p>
      * When <code>accept</code> is invoked on the proxy instance, context is first established on the thread that will
      * run the <code>accept</code> method, then the <code>accept</code> method of the provided <code>Consumer</code> is
@@ -516,7 +516,7 @@ public interface ThreadContext {
      * Wraps a <code>Function</code> with context that is captured from the thread that invokes
      * <code>contextualFunction</code>.
      * </p>
-     * 
+     *
      * <p>
      * When <code>apply</code> is invoked on the proxy instance, context is first established on the thread that will
      * run the <code>apply</code> method, then the <code>apply</code> method of the provided <code>Function</code> is
@@ -541,13 +541,13 @@ public interface ThreadContext {
      * Wraps a <code>Runnable</code> with context that is captured from the thread that invokes
      * <code>ContextualRunnable</code>.
      * </p>
-     * 
+     *
      * <p>
      * When <code>run</code> is invoked on the proxy instance, context is first established on the thread that will run
      * the <code>run</code> method, then the <code>run</code> method of the provided <code>Runnable</code> is invoked.
      * Finally, the previous context is restored on the thread, and control is returned to the invoker.
      * </p>
-     * 
+     *
      * @param runnable
      *            instance to contextualize.
      * @return contextualized proxy instance that wraps execution of the <code>run</code> method with context.
@@ -560,7 +560,7 @@ public interface ThreadContext {
      * <p>
      * Wraps a <code>Supplier</code> with context captured from the thread that invokes <code>contextualSupplier</code>.
      * </p>
-     * 
+     *
      * <p>
      * When <code>supply</code> is invoked on the proxy instance, context is first established on the thread that will
      * run the <code>supply</code> method, then the <code>supply</code> method of the provided <code>Supplier</code> is
@@ -590,7 +590,7 @@ public interface ThreadContext {
      * {@link ContextManager.Builder#withDefaultExecutorService(ExecutorService)}), or otherwise have no default
      * executor.
      * </p>
-     * 
+     *
      * <p>
      * If this thread context has no default executor, the new stage and all dependent stages created from it, and so
      * forth, have no default asynchronous execution facility and must raise
@@ -634,7 +634,7 @@ public interface ThreadContext {
      * {@link ContextManager.Builder#withDefaultExecutorService(ExecutorService)}), or otherwise have no default
      * executor.
      * </p>
-     * 
+     *
      * <p>
      * If this thread context has no default executor, the new stage and all dependent stages created from it, and so
      * forth, have no default asynchronous execution facility and must raise

--- a/api/src/main/java/org/eclipse/microprofile/context/spi/ContextManager.java
+++ b/api/src/main/java/org/eclipse/microprofile/context/spi/ContextManager.java
@@ -56,7 +56,7 @@ public interface ContextManager {
 
         /**
          * Use the specified {@link ThreadContextProvider} instances.
-         * 
+         *
          * @param providers
          *            the {@link ThreadContextProvider} instances to use.
          * @return this builder
@@ -67,14 +67,14 @@ public interface ContextManager {
          * Load all discoverable {@link ContextManagerExtension} instances via the {@link java.util.ServiceLoader}
          * mechanism on the current thread-context {@link ClassLoader} (unless overridden by
          * {@link #forClassLoader(ClassLoader)}).
-         * 
+         *
          * @return this builder
          */
         public ContextManager.Builder addDiscoveredContextManagerExtensions();
 
         /**
          * Use the specified {@link ContextManagerExtension} instances.
-         * 
+         *
          * @param extensions
          *            the {@link ContextManagerExtension} instances to use.
          * @return this builder
@@ -85,7 +85,7 @@ public interface ContextManager {
          * Load all discoverable {@link ThreadContextProvider} instances via the {@link java.util.ServiceLoader}
          * mechanism on the current thread-context {@link ClassLoader} (unless overridden by
          * {@link #forClassLoader(ClassLoader)}).
-         * 
+         *
          * @return this builder
          */
         public Builder addDiscoveredThreadContextProviders();
@@ -93,7 +93,7 @@ public interface ContextManager {
         /**
          * Use the given {@link ClassLoader} for {@link #addDiscoveredThreadContextProviders()} instead of the current
          * thread-context {@link ClassLoader}.
-         * 
+         *
          * @param classLoader
          *            the {@link ClassLoader} to use for {@link #addDiscoveredThreadContextProviders()}
          * @return this builder
@@ -103,7 +103,7 @@ public interface ContextManager {
         /**
          * Use the given {@link ExecutorService} to execute async tasks for contextualised {@link CompletableFuture} and
          * {@link CompletionStage} when no executor is specified.
-         * 
+         *
          * @param executorService
          *            the {@link ExecutorService} to use for async tasks when no executor is specified.
          * @return this builder
@@ -114,12 +114,12 @@ public interface ContextManager {
          * <p>
          * Creates a new {@link ContextManager} with the specified configuration.
          * </p>
-         * 
+         *
          * <p>
          * Creating a {@link ContextManager} will load and invoke all related {@link ContextManagerExtension} as
          * described in its documentation.
          * </p>
-         * 
+         *
          * @return a new {@link ContextManager} with the specified configuration.
          */
         public ContextManager build();

--- a/api/src/main/java/org/eclipse/microprofile/context/spi/ContextManagerProvider.java
+++ b/api/src/main/java/org/eclipse/microprofile/context/spi/ContextManagerProvider.java
@@ -35,7 +35,7 @@ import java.util.concurrent.atomic.AtomicReference;
  * The container must register its <code>ContextManagerProvider</code> implementation via the <code>register</code>
  * method, or by providing an implementation via the standard {@link ServiceLoader} mechanism.
  * </p>
- * 
+ *
  * <p>
  * <code>ContextManagerProvider</code> implementations that wish to use the {@link ServiceLoader} registration mechanism
  * must include a file of the following name and location in their jar:
@@ -47,7 +47,7 @@ import java.util.concurrent.atomic.AtomicReference;
  * The content of the aforementioned file must be exactly one line, specifying the fully qualified name of a
  * <code>ContextManagerProvider</code> implementation that is provided within the JAR file.
  * </p>
- * 
+ *
  * <p>
  * If there is no manually registered <code>ContextManagerProvider</code> (via
  * {@link #register(ContextManagerProvider)}), any call to {@link #instance()} will look up any
@@ -110,7 +110,7 @@ public interface ContextManagerProvider {
      * Gets a {@link ContextManager} for the current thread-context {@link ClassLoader}. This is equivalent to calling
      * <code>getContextManager(Thread.currentThread().getContextClassLoader())</code>, which is the default
      * implementation of this method.
-     * 
+     *
      * @return a {@link ContextManager} for the current thread-context {@link ClassLoader}.
      * @throws IllegalStateException
      *             if more than one {@link ThreadContextProvider} provides the same thread context
@@ -153,7 +153,7 @@ public interface ContextManagerProvider {
      * Returns a new {@link ContextManager.Builder} to create new {@link ContextManager} instances. Watch out that
      * instances created this way will not be automatically registered here, so you need to call
      * {@link #registerContextManager(ContextManager, ClassLoader)} yourself if you need to.
-     * 
+     *
      * @return a new {@link ContextManager.Builder}
      * @throws UnsupportedOperationException
      *             if the <code>ContextManagerProvider</code> always uses the same set of
@@ -167,7 +167,7 @@ public interface ContextManagerProvider {
      * Registers the given {@link ContextManager} for the given {@link ClassLoader}, so that further calls to
      * {@link #getContextManager(ClassLoader)} for the same {@link ClassLoader} will return this instance instead of
      * creating a new one.
-     * 
+     *
      * @param manager
      *            The {@link ContextManager} to register
      * @param classLoader
@@ -185,7 +185,7 @@ public interface ContextManagerProvider {
     /**
      * Releases a {@link ContextManager} that was previously registered with
      * {@link #registerContextManager(ContextManager, ClassLoader)}.
-     * 
+     *
      * @param manager
      *            The {@link ContextManager} to release
      * @throws UnsupportedOperationException

--- a/api/src/main/java/org/eclipse/microprofile/context/spi/ThreadContextController.java
+++ b/api/src/main/java/org/eclipse/microprofile/context/spi/ThreadContextController.java
@@ -43,7 +43,7 @@ public interface ThreadContextController {
      * <p>
      * Typically, patterns such as the following will be observed:
      * </p>
-     * 
+     *
      * <pre>
      * <code>
      * controllerA1 = contextSnapshotA.begin();
@@ -59,7 +59,7 @@ public interface ThreadContextController {
      * <p>
      * However, more advanced sequences such as the following are also valid:
      * </p>
-     * 
+     *
      * <pre>
      * <code>
      * controllerA1 = contextSnapshotA.begin();

--- a/api/src/main/java/org/eclipse/microprofile/context/spi/ThreadContextProvider.java
+++ b/api/src/main/java/org/eclipse/microprofile/context/spi/ThreadContextProvider.java
@@ -128,7 +128,7 @@ public interface ThreadContextProvider {
      * <p>
      * For example:
      * </p>
-     * 
+     *
      * <pre>
      * <code>
      * ManagedExecutor executor = ManagedExecutor.builder()

--- a/pom.xml
+++ b/pom.xml
@@ -18,7 +18,7 @@
     <parent>
         <groupId>org.eclipse.microprofile</groupId>
         <artifactId>microprofile-parent</artifactId>
-        <version>2.5</version>
+        <version>3.4</version>
     </parent>
 
     <groupId>org.eclipse.microprofile.context-propagation</groupId>
@@ -31,7 +31,41 @@
 
     <properties>
         <inceptionYear>2018</inceptionYear>
+        <!--  Dependencies - Test  -->
+        <version.testng>7.4.0</version.testng>
+        <!--  aligns with Arquillian TestNG version  -->
+        <version.junit>4.13.1</version.junit>
+        <version.hamcrest>1.3</version.hamcrest>
+        <version.arquillian>1.7.0.Alpha9</version.arquillian>
     </properties>
+
+    <dependencyManagement>
+        <dependencies>
+            <!--  Test Dependencies  -->
+            <dependency>
+                <groupId>org.testng</groupId>
+                <artifactId>testng</artifactId>
+                <version>${version.testng}</version>
+            </dependency>
+            <dependency>
+                <groupId>junit</groupId>
+                <artifactId>junit</artifactId>
+                <version>${version.junit}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.hamcrest</groupId>
+                <artifactId>hamcrest-all</artifactId>
+                <version>${version.hamcrest}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.jboss.arquillian</groupId>
+                <artifactId>arquillian-bom</artifactId>
+                <version>${version.arquillian}</version>
+                <scope>import</scope>
+                <type>pom</type>
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
 
     <issueManagement>
         <system>GitHub</system>

--- a/pom.xml
+++ b/pom.xml
@@ -31,6 +31,8 @@
 
     <properties>
         <inceptionYear>2018</inceptionYear>
+        <maven.compiler.source>9</maven.compiler.source>
+        <maven.compiler.target>9</maven.compiler.target>
         <!--  Dependencies - Test  -->
         <version.testng>7.4.0</version.testng>
         <!--  aligns with Arquillian TestNG version  -->

--- a/tck/src/main/java/org/eclipse/microprofile/context/tck/ManagedExecutorTest.java
+++ b/tck/src/main/java/org/eclipse/microprofile/context/tck/ManagedExecutorTest.java
@@ -1049,7 +1049,7 @@ public class ManagedExecutorTest extends Arquillian {
      * all dependent stages of the completed future that is created by the ManagedExecutor's failedFuture
      * implementation. Thread context is captured at each point where a dependent stage is added, rather than solely
      * upon creation of the initial stage or construction of the builder.
-     * 
+     *
      * @throws ExecutionException
      *             indicates test failure
      * @throws InterruptedException
@@ -2828,7 +2828,7 @@ public class ManagedExecutorTest extends Arquillian {
 
     /**
      * Verify that we can copy a CompletableFuture and get context propagation in the copy's dependent stages.
-     * 
+     *
      * @throws InterruptedException
      *             indicates test failure
      * @throws ExecutionException
@@ -2874,7 +2874,7 @@ public class ManagedExecutorTest extends Arquillian {
 
     /**
      * Verify that we can copy a CompletionStage and get context propagation in the copy's dependent stages.
-     * 
+     *
      * @throws InterruptedException
      *             indicates test failure
      * @throws ExecutionException
@@ -2920,7 +2920,7 @@ public class ManagedExecutorTest extends Arquillian {
 
     /**
      * Verify that we can obtain a ThreadContext with the same settings as the ManagedExecutor
-     * 
+     *
      * @throws InterruptedException
      *             indicates test failure
      * @throws ExecutionException

--- a/tck/src/main/java/org/eclipse/microprofile/context/tck/contexts/label/Label.java
+++ b/tck/src/main/java/org/eclipse/microprofile/context/tck/contexts/label/Label.java
@@ -32,7 +32,7 @@ public class Label {
 
     /**
      * Get the current 'label' context.
-     * 
+     *
      * @return current label.
      */
     public static String get() {
@@ -41,7 +41,7 @@ public class Label {
 
     /**
      * Set the current 'label' context.
-     * 
+     *
      * @param label
      *            new label.
      */

